W-14137751-try-scope-cv (#177)
mulesoft/docs-code-builder: modules/ROOT/pages/_partials/mule-components/4.4/component-try.adoc

//
//tag::component-title[]

= Try Scope (<try/>)

//end::component-title[]
//

//
//tag::component-short-description[]
//     Short description of the form "Do something..." 
//     Example: "Configure log messages anywhere in a flow."

> Handle errors that may occur when attempting to execute any of the components inside the scope.

Reviewer: ```suggestion
Handle errors that might occur during the execution of components inside the scope.
```

//end::component-short-description[]
//

//
//tag::component-long-description[]

> A Try scope wraps one or more operations, then catches and handles any exceptions that might be thrown by any of these enclosed operations. The behavior is as if you extracted those enclosed event components into a separate flow with its own error handling strategy, but inline, without having to actually define a new flow.

Reviewer: ```suggestion
The Try scope catches and handles exceptions thrown by the components that you add to the scope. Error handling within the scope is similar to error handling within a Flow component. Instead of configuring a separate flow, you can use Try within a Flow component. 
```

Reviewer: Might be nice to reword "The behavior is as if you" and " but inline, without having to actually define a new flow," but I don't have great suggestions. I'm not sure if "inline" is 100% clear here. 

The scope contains its own components and error handling strategy, similar to a flow. ¯\_(ツ)_/¯

Every payload modification or variable addition, modification, or removal is propagated through the rest of the execution. This propagation includes modifications that take place inside the error handlers.

> The Try scope supports transactions. A transaction is a series of actions that should never be partially executed. Configure a Try scope so that it is a set of operations that are considered one unit that either succeeds or fails, depending on whether errors are propagated and the transaction rolled back, or handled and the transaction committed. In either case, the process flow in which the Try scope resides continues.

Reviewer: ```suggestion
The Try scope supports transactions. A transaction is a series of actions that must execute successfully. Configure components within the Try scope as a single unit that either succeeds or fails, depending on whether errors are propagated and the transaction rolled back, or handled and the transaction committed. In either case, the flow that contains the Try scope continues.
```

Reviewer: I'm not sure what "process flow" means here.
